executor/oci/resolvconf_test.go

```diff
@@ -2,34 +2,150 @@ package oci
+		execution          int
+		hostNetMode        []bool
+		expected           []string
+	}{
+		{
+			name:               "TestResolvConfNotExist",
+			resolvconfGetFile:  nil,
+			resolvconfGetError: os.ErrNotExist,
+			execution:          1,
+			hostNetMode:        []bool{false},
+			expected:           []string{defaultResolvConf},
+		},
+		{
+			name:               "TestNetModeIsHostResolvConfNotExist",
+			resolvconfGetFile:  nil,
+			resolvconfGetError: os.ErrNotExist,
+			execution:          1,
+			hostNetMode:        []bool{true},
+			expected:           []string{defaultResolvConf},
+		},
+		{
+			name: "TestNetModeIsHostWithoutLocalDNS",
+			resolvconfGetFile: &resolvconf.File{
+				Content: []byte(regularResolvConf),
+			},
+			resolvconfGetError: nil,
+			execution:          1,
+			hostNetMode:        []bool{true},
+			expected:           []string{regularResolvConf},
+		},
+		{
+			name: "TestNetModeIsHostWithLocalDNS",
+			resolvconfGetFile: &resolvconf.File{
+				Content: []byte(localDNSResolvConf),
+			},
+			resolvconfGetError: nil,
+			execution:          1,
+			hostNetMode:        []bool{true},
+			expected:           []string{localDNSResolvConf},
+		},
+		{
+			name: "TestNetModeNotHostWithoutLocalDNS",
+			resolvconfGetFile: &resolvconf.File{
+				Content: []byte(regularResolvConf),
+			},
+			resolvconfGetError: nil,
+			execution:          1,
+			hostNetMode:        []bool{false},
+			expected:           []string{regularResolvConf},
+		},
+		{
+			name: "TestNetModeNotHostWithLocalDNS",
+			resolvconfGetFile: &resolvconf.File{
+				Content: []byte(localDNSResolvConf),
+			},
+			resolvconfGetError: nil,
+			execution:          1,
+			hostNetMode:        []bool{false},
+			expected:           []string{fmt.Sprintf("%s%s", dnsOption, defaultResolvConf)},
+		},
+		{
+			name: "TestRegenerateResolvconfToRemoveLocalDNS",
+			resolvconfGetFile: &resolvconf.File{
+				Content: []byte(localDNSResolvConf),
+			},
+			resolvconfGetError: nil,
+			execution:          2,
+			hostNetMode:        []bool{true, false},
+			expected: []string{
+				localDNSResolvConf,
+				fmt.Sprintf("%s%s", dnsOption, defaultResolvConf),
+			},
+		},
+		{
+			name: "TestRegenerateResolvconfToAddLocalDNS",
+			resolvconfGetFile: &resolvconf.File{
+				Content: []byte(localDNSResolvConf),
+			},
+			resolvconfGetError: nil,
+			execution:          2,
+			hostNetMode:        []bool{false, true},
+			expected: []string{
+				fmt.Sprintf("%s%s", dnsOption, defaultResolvConf),
+				localDNSResolvConf,
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			oldResolvconfGet := resolvconfGet
+			defer func() {
+				resolvconfGet = oldResolvconfGet
+			}()
+			resolvconfGet = func() (*resolvconf.File, error) {
+				return tc.resolvconfGetFile, tc.resolvconfGetError
+			}
+
+			ctx := context.Background()
+			tempDir := t.TempDir()
+
+			for i := 0; i < tc.execution; i++ {
+				if i > 0 {
+					time.Sleep(100 * time.Millisecond)
```

Comment:
The "TestRegenerateResolvconf*" tests are flaky without `time.Sleep`... any ideas for not using sleep?